interfaces/cython/cantera/onedim.py

```diff
@@ -639,6 +622,28 @@ def E(self):
         Efield.append((phi[np-2] - phi[np-1]) / (z[np-1] - z[np-2]))
         return Efield
 
+    def solve(self, loglevel=1, refine_grid=True, auto=False, stage=1, enable_energy=True):
+        self.flame.set_solvingStage(stage)
+        if stage == 1:
+            super(IonFlameBase, self).solve(loglevel, refine_grid, auto)
+        if stage == 2:
+            self.poisson_enabled = True
+            super(IonFlameBase, self).solve(loglevel, refine_grid, auto)
+
+
+class IonFlame(IonFlameBase, FreeFlame):
+    __slots__ = ('inlet', 'outlet', 'flame')
+
+    def __init__(self, gas, grid=None, width=None):
+        if not hasattr(self, 'flame'):
+            # Create flame domain if not already instantiated by a child class
+            self.flame = IonFreeFlow(gas, name='flame')
+
+        super(IonFlame, self).__init__(gas, grid, width)
+
+    def solve(self, loglevel=1, refine_grid=True, auto=False, stage=1, enable_energy=True):
+        IonFlameBase.solve(self, loglevel, refine_grid, auto, stage, enable_energy)
```

Comment:
This method is unnecessary -- since `IonFlameBase` is the first base class listed, this class's `solve` method is the one that will be called.